### DISCUSSION TITLE
Change "You have looked up this player recently" to a more appropriate status code

### DISCUSSION THREAD

**Discussion by Clemintina** [HypixelDev/PublicAPI#566] [Ideas]
Hey!

I've noticed the **/player** endpoint responds with 429 ( too many requests ) when you request the same player twice. This status code is normally reserved for api keys ( rate-limited ) and is stated as such in the Mozilla docs.  

My proposal is to change the code from 429 to 400 which is more suited to the response. Since requesting a player twice using name is considered a bad request not being rate-limited.

**Comment by ptlthg** [comment]
Are you able to reproduce this? I just requested the same player 10 times in a row without issue. Are you sure that you didn't just hit your rate limit?

If not, I agree that `429` could be confusing, however the proposal of `400` doesn't sit right with me either. A `400: Bad Request` is typically reserved for an actual malformed or improper request and is telling the client not to repeat it. `425: Too Early` could work, but that also seems a bit too specific about replaying resources. 

For fun, `420: Enchance Your Calm` would be amusing, but I'd understand if that one is skipped (Even though I'd argue it's most appropriate).

**Reply by Clemintina** [reply]
> Are you able to reproduce this? I just requested the same player 10 times in a row without issue. Are you sure that you didn't just hit your rate limit?
> 
> If not, I agree that `429` could be confusing, however the proposal of `400` doesn't sit right with me either. A `400: Bad Request` is typically reserved for an actual malformed or improper request and is telling the client not to repeat it. `425: Too Early` could work, but that also seems a bit too specific about replaying resources.
> 
> For fun, `420: Enchance Your Calm` would be amusing, but I'd understand if that one is skipped (Even though I'd argue it's most appropriate).

This is for the ``name`` parameter. You can request the same ``uuid`` without any issues.

**Reply by ptlthg** [reply]
Ah, I see now. Thanks for the clarification. It was my understanding that you shouldn't use the `name` parameter at all in favor of fetching UUIDs from the Mojang API to then use. (Note that the `name` parameter isn't even listed on the API specification anymore)

Because of that, I don't think changing the status code here matters at all. On your end, you could also read the returned JSON error cause.
```json
{
    "success": false,
    "cause": "You have already looked up this name recently"
}
```

**Reply by Clemintina** [reply]
> Ah, I see now. Thanks for the clarification. It was my understanding that you shouldn't use the `name` parameter at all in favor of fetching UUIDs from the Mojang API to then use. (Note that the `name` parameter isn't even listed on the API specification anymore)
> 
> Because of that, I don't think changing the status code here matters at all. On your end, you could also read the returned JSON error cause.
> 
> ```json
> {
>     "success": false,
>     "cause": "You have already looked up this name recently"
> }
> ```

I agree you should use the ``uuid`` parameter however it's just better to change the code to a more appropriate code just for standardisation and clarity. Some people don't have the luxury of using uuids which is why the name parameter hasn't been fully deprecated yet as it's supposed to be used as a fallback. 

Mojang's rate limit is also pretty low and playerdb has a rate limit even though it's unstated. So using the name parameter on Hypixel makes more sense for people with those limitations.
